### DISCUSSION TITLE
not really the fault of rich, but maybe rich can help...

### DISCUSSION THREAD

**Discussion by franzhaas** [Textualize/rich#3124] [Show and tell]
Not really a bug of rich... 

I messed up and had nbconvert installed globally and used the globally installed jupyter-nbconvert to convert ipynb files which I did create with a virtualenv to revearl.js presentations.

This worked surprisingly well until I started to use rich with progress bars, these are then not rendered correctly in the resulting presentation, although the rest is rendered as expected.

This took me quite a while to figure out. An error message, an entry in the FAQ or something like that might help here.

Ideally an error message on the console/in the HTML would be great, or an entry in the documentation. 
This really did puzzle me for a while.

Thanks for the great tool!

**Comment by willmcgugan** [comment]
I can't offer any suggestions until you share what the problem is...

**Comment by franzhaas** [comment]
Ah sorry, the symptom... is.:

when using the out of virtual environment nbconvert the rendered result looks like this.:
![image](https://github.com/Textualize/rich/assets/11477153/5c705e2a-3ad7-45f8-aed3-0e6eb181d09a)

when installing nbconvert into the virutalenv the result looks like this.:

![image](https://github.com/Textualize/rich/assets/11477153/8d12ea14-f800-4271-9dbb-03d9c7ccf539)

I believe this is due to the fact that nbconvert needs something from rich (or a dependency thereof) while running, which is available in the virtual environment but not in the global environment.

This should not happen, as having nbconvert in the same python installation as rich is clearly the correct solution, however, when this mistake is made, a helpful entry in the documentation/FAQ, or an error message would be helpful.

The text I ended up to google without success was "<pre style="white-space:pre;overflow-x:auto;line-height:normal;.

Thanks,
Franz

**Reply by willmcgugan** [reply]
I maintain Rich, but I have nothing to do with nbconvert. Your first port of call should be the developers behind nbconvert.

**Reply by franzhaas** [reply]
Thanks... I believe they will send me off to the virtual environment tool, which is the place where I intend to fix this issue permanently.

The reason I (by mistake) believed that rich is at fault, is that all non-rich related renderings looked so good. -> Therefore my mind tricked itself into believing everything else is OK. -> This is my motivation to document this symptom here.

The problem is not at all limited/related to nbconvert or rich, and I am looking into solutions to avoid out of virtualenv calls for every and all reasons.

Thanks again,
Franz
